main.go

```diff
@@ -71,6 +72,23 @@ func main() {
 	kongplete.Complete(parser, kongplete.WithPredictor("file", complete.PredictFiles("*")))
 
 	kongCtx, err := parser.Parse(os.Args[1:])
+	if err != nil {
+		var parseErr *kong.ParseError
+		if errors.As(err, &parseErr) {
+			// do not error on missing command/argument.
+			// Print Usage + friendly message instead.
+			if parseErr.Context.Error == nil {
+				node := parseErr.Context.Selected()
+				if node == nil {
+					node = parseErr.Context.Model.Node
+				}
+				if format.MissingChildren(node) {
+					err = format.NoFatalExitIfErrorf(err, parseErr.Context.Command())
+				}
+			}
+		}
+	}
+
 	parser.FatalIfErrorf(err)
```

Comment:
Nit: this could be moved up into the `if err != nil` block or maybe even removed? Is there any case where this is still needed?

Reply:
moving this up as suggested.

yes we need it - we still error on unknown argument, e.g.:
```bash
> nctl dummy
...
nctl: error: unexpected argument dummy
```